### PR TITLE
Adding hack to allow context-local safety for the pubsub client.

### PR DESCRIPTION
**Pull request by jonparrott**

### CODE REVIEW

**Comment by jonparrott** [comment]
See [gcloud-python#926](https://github.com/GoogleCloudPlatform/gcloud-python/issues/926)

File: psq/context_local_pubsub_connection_test.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from unittest import TestCase
+
+from mock import Mock
+
+from psq.context_local_pubsub_connection import ContextLocalPubsubConnection
+
+
```
Comment (waprin) [inline psq/context_local_pubsub_connection_test.py]: feel like there should be a test that spins up a second thread and verifies it gets a new client (first thread creates ContextLocal, start second thread ,that asserts there is no contextlocal connection and then creates one). Something like that...

Comment (jonparrott) [inline psq/context_local_pubsub_connection_test.py]: It's really hard to test that and I feel pretty comfortable that werkzeug's local does what it says.

**Comment by waprin** [comment]
lgtm

File: psq/context_local_pubsub_connection_test.py (hunk 1 above)
Comment (waprin) [inline psq/context_local_pubsub_connection_test.py]: fair enough

Comment (jonparrott) [inline psq/context_local_pubsub_connection_test.py]: For the record, I feel like that is a bad argument, but this is hopefully a temporary hack until gcloud-python has a more thorough thread-safety policy.

Comment (waprin) [inline psq/context_local_pubsub_connection_test.py]: Yeah, testing multi-threaded code is a headache. I don't think the test I suggested is too big a deal to write , but I also agree its' testing werkzeug more than your own code . But sometimes testing dependencies is worth it, ties into our gcloud testing debate ha.

**Review by Nutza97 — APPROVED** [review]

**Pull request merged**